crates/sui-core/src/node_sync/node_follower.rs

```diff
@@ -584,6 +585,9 @@ where
             });
         }
 
+        // Lets notify here waiters, to clean up resources
+        let _ = self.pending_downloads.notify(tx_digest, Ok(())).await;
```

Comment:
I think a better fix is to make the notify() call on line 352 unconditional - it currently does not fire in the error case.

Adding the call here doesn't hurt, but note that it can be missed due to early returns above.